web/packages/extension/src/utils.ts

```diff
@@ -184,3 +91,32 @@ export async function getExplicitOptions(): Promise<Options> {
 
     return options;
 }
+
+export const hasAllUrlsPermission = async () => {
+    const allPermissions = await permissions.getAll();
+    if (Object.prototype.hasOwnProperty.call(allPermissions, "origins")) {
+        return (
+            allPermissions.origins &&
+            allPermissions.origins.indexOf("<all_urls>") > -1
+        );
+    }
+    return false;
+};
+
+export async function hasHostPermissionForActiveTab() {
+    const [activeTab] = await tabs.query({
+        active: true,
+        currentWindow: true,
+    });
+
+    try {
+        return activeTab?.url
```

Comment:
I was under the impression that having access the `url` property would imply a host permission?

Reply:
This was actually code I sort of swiped: https://discourse.mozilla.org/t/detect-whether-extension-has-host-permission-for-active-tab/120501/3. I guess it was unnecessarily complex.

Reply:
Yeah, that was only needed with the tabs permission, which we don't grant.

Reply:
Though as noted in popup.ts, `activeTab.url` returns `undefined` on Chrome as it requires the `tabs` permission, so my else is needed.

Reply:
Reverted this back, as Firefox lets you access the current tab's URL at all times.